Recordings-OO-MVC/app/src/main/java/com/madappgang/recordings/views/PlayerView.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2018 MadAppGang.
+ *
+ * Created by Andrii Fedorov afedorov@madappgang.com on 6/13/18.
+ */
+
+package com.madappgang.recordings.views
+
+import android.content.Context
+import android.support.v7.widget.AppCompatButton
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import android.widget.SeekBar
+import android.widget.TextView
+import com.madappgang.recordings.R
+import com.madappgang.recordings.extensions.formatMilliseconds
+import com.madappgang.recordings.kit.Player
+
+internal class PlayerView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+    defStyleRes: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr, defStyleRes) {
+
+    var onStartPausePlayer: () -> Unit = {}
+    var onProgressChanged: (Int) -> Unit = {}
+
+    private val playerPosition by lazy { findViewById<TextView>(R.id.playerPosition) }
+    private val trackDuration by lazy { findViewById<TextView>(R.id.trackDuration) }
+    private val startPausePlayer by lazy { findViewById<AppCompatButton>(R.id.startPausePlayer) }
+    private val playerSeekBar by lazy { findViewById<SeekBar>(R.id.playerProgress) }
+
+    private var state: Player.State = Player.State.NOT_STARTED
+
+    init {
+        inflate(getContext(), R.layout.view_player, this)
+
+        init()
+        setState(Player.State.NOT_STARTED)
+        setTrackDuration(0)
+        setCurrentPosition(0)
+    }
+
+    fun setState(state: Player.State) {
+        this.state = state
+        updateStartPauseButton()
```

Comment:
You have no reason to **store** state in `PlayerView`. Just pass is through `updateStartPauseButton()` or other funcs.